edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Edgelet/versioning/ModuleManagementHttpClientVersioned.cs

```diff
@@ -103,6 +108,12 @@ public virtual async Task<Stream> GetModuleLogs(string module, bool follow, Opti
 
         protected abstract void HandleException(Exception ex, string operation);
 
+        bool IsModuleManagementSocketStale(Exception ex)
+        {
+            return (string.Compare(ex.GetType().ToString(), "System.Net.Internals.SocketExceptionFactory+ExtendedSocketException") == 0)
```

Comment:
catch socket exception first and check exception message. https://stackoverflow.com/questions/56909680/how-can-i-catch-an-extendedsocketexception. Also please ignore case when string comparison.

Also does .net core 3.1 throw the same exception?  Andrew's .net core 3.1 upgrade PR will be committed today.

Reply:
I don't know what exception Dotnet3.1 is going to throw. I'll have to test it out.. Can you point me to the PR please? I'd like to get the testing going asap. 

I certainly can also added the `if(ex is SocketException)`